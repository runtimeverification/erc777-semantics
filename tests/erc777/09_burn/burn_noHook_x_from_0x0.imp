test.preamble();

// burn(),  operatorBurn() through default operator (unless otherwise specified), operatorBurn() from caller.
// holder is 0x0, operatorBurn() from authorized op, burning some balance. Should accept, burning from operator.

test.makeCaller(8);
print("Actual burner: ");
print(8);
print("\nBalance of ");
print(8);
print(" is ");
println(balanceOf(8));
test.operatorBurn(0, 3, ""); //through default operator
print("Balance of ");
print(8);
print(" is ");
println(balanceOf(8));
