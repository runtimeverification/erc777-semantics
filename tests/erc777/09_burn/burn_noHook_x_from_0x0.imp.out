Changing the caller to 8
Actual burner: 8
Balance of 8 is 80
8 is burning 3 units from 0 (operatorData:"")
Balance of 0 is 0
Total supply is 547
Balance of 8 is 77
